pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c

```diff
@@ -0,0 +1,666 @@
+{
+    LogInfo("Proccesing Property Update: %s", PropertyName);
+    LogInfo("   New Property Value: %s", json_value_get_string(PropertyValue));
+}
+
+void ImpinjReader_OnPropertyCallback(
+    PNPBRIDGE_COMPONENT_HANDLE PnpComponentHandle,
+    const char* PropertyName,
+    JSON_Value* PropertyValue,
+    int version,
+    void* ClientHandle
+)
+// {
+//     SampleEnvironmentalSensor_ProcessPropertyUpdate(userContextCallback, PropertyName, PropertyValue, version, PnpComponentHandle);
+// }
+
+// void SampleEnvironmentalSensor_ProcessPropertyUpdate(
+//     void * ClientHandle,
+//     const char* PropertyName,
+//     JSON_Value* PropertyValue,
+//     int version,
+//     PNPBRIDGE_COMPONENT_HANDLE PnpComponentHandle)
+{
+    
+    LogInfo("Processing Property: %s", PropertyName);
+    if (strcmp(PropertyName, impinjReader_property_testPropertyString) == 0)
+    {
+        ImpinjReader_ProcessProperty(ClientHandle, PropertyName, PropertyValue, version, PnpComponentHandle);
+    }
+    // if (strcmp(PropertyName, sampleEnvironmentalSensorPropertyCustomerName) == 0)
+    // {
+    //     SampleEnvironmentalSensor_CustomerNameCallback(ClientHandle, PropertyName, PropertyValue, version, PnpComponentHandle);
+    // }
+    // else if (strcmp(PropertyName, sampleEnvironmentalSensorPropertyBrightness) == 0)
+    // {
+    //     SampleEnvironmentalSensor_BrightnessCallback(ClientHandle, PropertyName, PropertyValue, version, PnpComponentHandle);
+    // }
+    // else if (strcmp(PropertyName, sampleDeviceStateProperty) == 0)
+    // {
+    //     const char * PropertyValueString = json_value_get_string(PropertyValue);
+    //     size_t PropertyValueLen = strlen(PropertyValueString);
+
+    //     LogInfo("Environmental Sensor Adapter:: Property name <%s>, last reported value=<%.*s>",
+    //         PropertyName, (int)PropertyValueLen, PropertyValueString);
+    // }
+    else
+    {
+        // If the property is not implemented by this interface, presently we only record a log message but do not have a mechanism to report back to the service
+        LogError("Impinj Reader Adapter:: Property name <%s> is not associated with this interface", PropertyName);
+    }
+}
+
+int ImpinjReader_ProcessCommand(
+    PIMPINJ_READER ImpinjReader,
+    const char* CommandName,
+    JSON_Value* CommandValue,
+    unsigned char** CommandResponse,
+    size_t* CommandResponseSize)
+{
+    if (strcmp(CommandName, "startPreset") == 0)
+    {
+
+        char startPresetEndpoint_build[100] = "";
+        strcat(startPresetEndpoint_build, "/profiles/inventory/presets/");
+        strcat(startPresetEndpoint_build, json_value_get_string(CommandValue));
+        strcat(startPresetEndpoint_build, "/start");
+        char *startPresetEndpoint = Str_Trim(startPresetEndpoint_build).strPtr;
+
+        char* res = curlStaticPost(ImpinjReader->curl_static_session, startPresetEndpoint, "");
+
+        char * response = ImpinjReader_CreateJsonResponse("cmdResponse", res);
```

Comment:
>"cmdResponse", res); [](start = 58, length = 20)

From the model it looks like you're expecting a string schema inside this command response. This is probably something that can be cleanly set with a message format. If you expect the adapter to cater to different *types* of schemas for command responses, the _CreateJsonResponse could differentiate the "capability type" by reading component adapter config (This would need to get cached before-hand - possibly during _CreatePnpAdapter) and matched during _OnCommandUpdate for specific components.